src/PostSharp.Engineering.BuildTools/Utilities/ToolInvocationHelper.cs

```diff
@@ -221,6 +222,12 @@ public static class ToolInvocationHelper
                         }
                         else
                         {
+                            // If ToolInvocationRetry.Regex matches output that causes build failure, we will set the build to retry.
```

Comment:
Don't copy-paste, ever. Use a local function in this case, which could be named FilterProcessOutput.